### PR TITLE
feat(route): Latest Stories from National Geographic

### LINKED ISSUE #12592: 请求生成美国国家地理官网rss
### 类型

出行旅游

### 网站地址

https://www.nationalgeographic.com/

### 网站描述

美国国家地理

### 需要生成什么内容？

LATEST STORIES的最新文章

### 额外描述

最好是能抓取全文，谢谢。

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

### PR DESCRIPTION
**Pull request by miles170**
## 该 PR 相关 Issue / Involved Issue

Close #12592

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/nationalgeographic/latest-stories
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [x] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [x] 反爬/频率限制 anti-bot or rate limit?
  - [x] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

**Comment by jy53655** [comment]
感谢，但是还是用不了，提示如下：
Route requested: /nationalgeographic/latest-stories
Error message: wrong path
Helpful Information to provide when opening issue:
Path: /nationalgeographic/latest-stories
Node version: v18.16.0
Git Hash: 051a3ef

**Comment by miles170** [comment]
> 感谢，但是还是用不了，提示如下： Route requested: /nationalgeographic/latest-stories Error message: wrong path Helpful Information to provide when opening issue: Path: /nationalgeographic/latest-stories Node version: v18.16.0 Git Hash: 051a3ef

你需要等到这个 PR 被合併后才能够使用

File: lib/v2/nationalgeographic/latest-stories.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const asyncPool = require('tiny-async-pool');
+
+module.exports = async (ctx) => {
+    const currentUrl = 'https://www.nationalgeographic.com/pages/topic/latest-stories';
+    const response = await got(currentUrl);
+    const $ = cheerio.load(response.data);
+    const items = [];
+    for await (const data of asyncPool(5, $('article a[class="AnchorLink PromoTile__Link"]'), (item) => {
```
Comment (TonyRL) [inline lib/v2/nationalgeographic/latest-stories.js:11]: I tried changing it to 50 and didn't hit a rate limit at all.

Comment (TonyRL) [inline lib/v2/nationalgeographic/latest-stories.js:11]: Please use the embedded object `__natgeo__` instead.

File: lib/v2/nationalgeographic/latest-stories.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        item = $(item);
+        const link = item.attr('href');
+
+        return ctx.cache.tryGet(link, async () => {
+            const response = await got(link);
+            const $ = cheerio.load(response.data);
+            const article = $('article section.Article__Content');
```
Comment (TonyRL) [inline lib/v2/nationalgeographic/latest-stories.js:18]: Please use the embedded object `__natgeo__` instead.

<details><summary>Screenshot</summary>

image
</details>

File: lib/v2/nationalgeographic/latest-stories.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                    }
+                });
+                return {
+                    title: article.meta.ttl,
+                    author: bodyTile.schma.athrs.map((a) => a.name).join(','),
+                    description: $body.html(),
+                    category: article.meta.pgTxnmy.sources,
+                    pubDate: parseDate(bodyTile.pbDt),
+                    link,
+                };
```
Comment (TonyRL) [inline lib/v2/nationalgeographic/latest-stories.js:79]: - You can make use of [`.find()`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/find)
- Use `art-template` for rendering custom HTML
```js
                const mods = findNatgeo($).page.content.article.frms.find((f) => f.cmsType === 'ArticleBodyFrame').mods;

                const bodyTile = mods.find((m) => m.edgs[0].cmsType === 'ArticleBodyTile').edgs[0];

                item.author = bodyTile.schma.athrs.map((a) => a.name).join(', ');
                item.description = art(join(__dirname, 'templates/stories.art'), {
                    ldMda: bodyTile.ldMda,
                    description: bodyTile.dscrptn,
                    body: bodyTile.bdy,
                });
                item.pubDate = parseDate(bodyTile.pbDt);
```
```mustache
{{ if ldMda }}
    <figure>
        <img src="{{ ldMda.image.src }}" alt="{{ ldMda.image.altText }}">
        <figcaption>{{ ldMda.caption }}</figcaption>
    </figure>
{{ /if }}

{{ if description }}
    <p><b>{{ dscrptn }}</b></p>
{{ /if }}

{{ each body b }}
    {{ if b.type === 'p' }}
        <p>{{@ b.cntnt.mrkup }}</p>
    {{ else if b.type === 'inline' }}
        {{ if b.cntnt.cmsType === 'image' }}
            <figure>
                <img src="{{ b.cntnt.image.src }}" alt="{{ b.cntnt.image.altText }}">
                <figcaption>{{@ b.cntnt.caption }}</figcaption>
            </figure>
        {{ else if b.cntnt.cmsType === 'imagegroup' }}
            {{ each b.cntnt.images img }}
                <figure>
                    <img src="{{ img.image.src }}" alt="{{ img.image.altText }}">
                    <figcaption>{{@ img.caption }}</figcaption>
                </figure>
            {{ /each }}
        {{ else if b.cntnt.cmsType === 'editorsNote' }}
            <p>{{@ b.cntnt.note }}</p>
        {{ else if b.cntnt.cmsType === 'listicle' }}
            <p>{{@ b.cntnt.text }}</p>
        {{ else if b.cntnt.cmsType === 'pullquote' }}
            <backquote>{{@ b.cntnt.quote }}</backquote>
        {{ else if b.cntnt.cmsType === 'source' }}
            <a href="{{ b.cntnt.src }}">{{ b.cntnt.src }}</a>
        {{ else if b.cntnt.cmsType === 'video' }}
            <figure>
                <img src="{{ b.cntnt.image.src }}" alt="{{ b.cntnt.image.altText }}">
                <figcaption>{{@ b.cntnt.title }}</figcaption>
                <figcaption>{{@ b.cntnt.description }}</figcaption>
            </figure>
        {{ /if }}
    {{ /if }}
{{ /each }}

```

**Pull request merged**